src/compiler.ts

```diff
@@ -10508,8 +10508,12 @@ export class Compiler extends DiagnosticEmitter {
       }
     }
     var parameterTypes = signature.parameterTypes;
+    var parameterNodes = reportNode.parameters;
     for (let i = 0, k = parameterTypes.length; i < k; ++i) {
-      if (!this.checkTypeSupported(parameterTypes[i], reportNode.parameters[i])) {
+      let parameterReportNode: Node;
+      if (parameterNodes.length > i) parameterReportNode = parameterNodes[i];
```

Comment:
Here, the function may come from a function expression with an omitted argument being compiled in context of a function type with fewer argument expressions, implicitly dropping excess arguments due to being irrelevant. Doesn't error when compiling to JS due to undefined, but produced a fine runtime index-out-of-bounds in Wasm.